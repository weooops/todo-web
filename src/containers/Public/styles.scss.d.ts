export const public: string;
export const text: string;
export const changeLink: string;
