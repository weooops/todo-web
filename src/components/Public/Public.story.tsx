import * as React from 'react';
import { Provider } from 'react-redux';
import { storiesOf } from '@storybook/react';

import configureStore from '../../configureStore';
import Public from './';

const store = configureStore();

storiesOf('Public', module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add('default', () => <Public />);
