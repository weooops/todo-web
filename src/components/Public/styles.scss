.auth {
  border: 1px solid red;
}