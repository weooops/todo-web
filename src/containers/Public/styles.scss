.public {
  // margin: 2px;
  // border: 2px solid red;
  max-width: 260px;
  width: 100%;

  .text {
    padding: 12px 0;
    text-align: center;
    color: #8a959e;

    .change-link {
      color: #fc5459;
      cursor: pointer;
    }
  }
}
