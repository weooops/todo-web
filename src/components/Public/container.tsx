import * as React from 'react';

import Public from './Public';

interface StateType {
  action: string;
}

class Container extends React.Component<{}, StateType> {
  public state: StateType = {
    action: 'login'
  };

  public render(): JSX.Element {
    const { action } = this.state;

    return (
      <Public
        action={action}
        onChangeAction={this._onChangeAction}
      />
    );
  }

  private _onChangeAction = (): void => {
    this.setState((prevState: StateType) => {
      const { action } = prevState;
      if (action === 'login') {
        return { action: 'signup' };
      } else {
        return { action: 'login' };
      }
    });
  }
}

export default Container;
