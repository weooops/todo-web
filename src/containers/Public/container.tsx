import * as React from 'react';

import Public from './Public';

interface IStates {
  action: string;
}

class Container extends React.Component<{}, IStates> {
  public state: IStates = {
    action: 'login'
  };

  public render(): JSX.Element {
    const { action } = this.state;

    return (
      <Public
        action={action}
        onChangeAction={this._onChangeAction}
      />
    );
  }

  private _onChangeAction = (): void => {
    this.setState((prevState: IStates) => {
      const { action } = prevState;
      if (action === 'login') {
        return { action: 'signup' };
      } else {
        return { action: 'login' };
      }
    });
  }
}

export default Container;
