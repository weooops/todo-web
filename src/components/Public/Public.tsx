import * as React from 'react';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import LoginForm from '../LoginForm';
import SignupForm from '../SignupForm';

const cx = classNames.bind(styles);

interface PropsType {
  action: string;
  onChangeAction(): void;
}

const Public: React.SFC<PropsType> = props => (
  <div className={cx('auth')}>
    {props.action === 'login' && (
      <div>
        <LoginForm />
        <p>
          <span>Don't have an account?{' '}</span>
          <span onClick={props.onChangeAction}>Sign up</span>
        </p>
      </div>
    )}
    {props.action === 'signup' && (
      <div>
        <SignupForm />
        <p>
          <span>Have an account?{' '}</span>
          <span onClick={props.onChangeAction}>Log in</span>
        </p>
      </div>
    )}
  </div>
);

export default Public;
