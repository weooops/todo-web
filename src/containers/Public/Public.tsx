import * as React from 'react';

import * as styles from './styles.scss';
import * as classNames from 'classnames/bind';
import LoginForm from '../LoginForm';
import SignupForm from '../SignupForm';

const cx = classNames.bind(styles);

interface IProps {
  action: string;
  onChangeAction(): void;
}

const Public: React.SFC<IProps> = props => (
  <div className={cx('public')}>
    {props.action === 'login' && (
      <div>
        <LoginForm />
        <p className={cx('text')}>
          <span>Don't have an Account?{' '}</span>
          <span className={cx('change-link')} onClick={props.onChangeAction}>Sign up</span>
        </p>
      </div>
    )}
    {props.action === 'signup' && (
      <div>
        <SignupForm />
        <p className={cx('text')}>
          <span>Already have an Account?{' '}</span>
          <span className={cx('change-link')} onClick={props.onChangeAction}>Log In</span>
        </p>
      </div>
    )}
  </div>
);

export default Public;
