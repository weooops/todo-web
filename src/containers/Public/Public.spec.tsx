import * as React from 'react';
import { shallow } from 'enzyme';

import Public from './Public';

it('renders correctly', () => {
  const props = {
    action: 'login',
    onChangeAction: jest.fn()
  };
  const wrapper = shallow(<Public {...props} />);
  expect(wrapper).toMatchSnapshot();
});
