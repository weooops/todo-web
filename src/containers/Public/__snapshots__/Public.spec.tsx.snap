// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Public
    action="login"
    onChangeAction={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Connect(Container) />
          <p>
            <span>
              Don't have an account?
               
            </span>
            <span
              onClick={[Function]}
            >
              Sign up
            </span>
          </p>
        </div>,
        false,
      ],
      "className": "public",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Container) />,
            <p>
              <span>
                Don't have an account?
                 
              </span>
              <span
                onClick={[Function]}
              >
                Sign up
              </span>
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  Don't have an account?
                   
                </span>,
                <span
                  onClick={[Function]}
                >
                  Sign up
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Don't have an account?",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Don't have an account?",
                  " ",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign up",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Sign up",
                "type": "span",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Connect(Container) />
            <p>
              <span>
                Don't have an account?
                 
              </span>
              <span
                onClick={[Function]}
              >
                Sign up
              </span>
            </p>
          </div>,
          false,
        ],
        "className": "public",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(Container) />,
              <p>
                <span>
                  Don't have an account?
                   
                </span>
                <span
                  onClick={[Function]}
                >
                  Sign up
                </span>
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Don't have an account?
                     
                  </span>,
                  <span
                    onClick={[Function]}
                  >
                    Sign up
                  </span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Don't have an account?",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Don't have an account?",
                    " ",
                  ],
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign up",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Sign up",
                  "type": "span",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
